packages/css-engine/src/core/to-value.ts

```diff
@@ -16,16 +16,17 @@ const assertUnreachable = (_arg: never, errorMessage: string) => {
 
 export const toValue = (
   value?: StyleValue,
-  options: ToCssOptions = defaultOptions
+  options: ToCssOptions = {}
 ): string => {
+  const { withFallback = true, getAssetPath } = options;
```

Comment:
hmm I wasn't planing on adding application-level logic to css-engine, is there a different way to do this outside the engine?

Reply:
Then we need to map style object. Though we probably already do this. And types with asset will be css engine specific.

Reply:
maybe we could pass some object with toString method and cssEngine would just stringify it without knowing any internals?

Reply:
And then we need to map asset to apply loader. So maybe even change the type from full asset to {id,path}

Reply:
is that bad?

Reply:
Let's see

Reply:
And we also gonna have 2 toValue because style panel heavily use one from css-engine but it will not be compatible with data from db.

Reply:
I like Bogdan solution its just some path converter, alternatives from my POV makes code less readable abd cause creation of client only types for styles, or splitting toValue logic

Reply:
Yep, just tried to add separate types. And it's huge amount of conversions and complexities.

Reply:
I am open to any suggestions that allow us to decouple css-engine from application

Reply:
you could add some sort of plugin/transformer in a generic way that would do this mapping

Reply:
Maybe more like this? No reliance on "id".
```
toValue(value, { resolveImagePath(imageValue): string }?
```